### PR TITLE
Remove any redundant `label` look-ups from query plan

### PR DESCRIPTION
**Pull request by aelred**

### CODE REVIEW

**Comment by aelred** [comment]
!rtg - failure in `grakn-dist` test with Cassandra timeout https://jenkins.grakn.ai/job/unit-test-pull-request-grakn-titan/4273/

File: grakn-graql/src/main/java/ai/grakn/graql/internal/gremlin/sets/LabelFragmentSet.java [hunk 1]
```diff
@@ -45,4 +49,44 @@ Var type() {
     TypeLabel label() {
         return label;
     }
+
+    /**
+     * Optimise away any redundant {@link LabelFragmentSet}s. A {@link LabelFragmentSet} is considered redundant if:
+     * <ol>
+     *   <li>It refers to a type that exists in the graph
+     *   <li>It is not associated with a user-defined variable
```
Comment (fppt) [inline grakn-graql/src/main/java/ai/grakn/graql/internal/gremlin/sets/LabelFragmentSet.java]: SERIOUS nitpick but only one of these `<li>` has a matching `</li>`

File: grakn-graql/src/main/java/ai/grakn/graql/internal/gremlin/sets/LabelFragmentSet.java [hunk 2]
```diff
@@ -45,4 +49,44 @@ Var type() {
...
+            Collection<EquivalentFragmentSet> fragmentSets, GraknGraph graph) {
+
+        if (fragmentSets.size() <= 1) return false;
+
+        Iterable<LabelFragmentSet> labelFragments =
+                EquivalentFragmentSets.fragmentSetOfType(LabelFragmentSet.class, fragmentSets)::iterator;
+
+        for (LabelFragmentSet labelSet : labelFragments) {
+
+            boolean hasUserDefinedVar = labelSet.type().isUserDefinedName();
```
Comment (fppt) [inline grakn-graql/src/main/java/ai/grakn/graql/internal/gremlin/sets/LabelFragmentSet.java]: You can inline this but I am guessing you didn't for the sake of readability

File: grakn-graql/src/main/java/ai/grakn/graql/internal/gremlin/sets/LabelFragmentSet.java (hunk 1 above)
Comment (aelred) [inline grakn-graql/src/main/java/ai/grakn/graql/internal/gremlin/sets/LabelFragmentSet.java]: I totally knew this and was just being lazy! Someday I'll fix it I'm sure.

File: grakn-graql/src/main/java/ai/grakn/graql/internal/gremlin/sets/LabelFragmentSet.java (hunk 2 above)
Comment (aelred) [inline grakn-graql/src/main/java/ai/grakn/graql/internal/gremlin/sets/LabelFragmentSet.java]: Yeah, I deliberately extracted them so they matched the rules in the Javadoc.

**Pull request merged**